src/main/java/com/enonic/lib/guillotine/ProcessHtmlHandler.java

```diff
@@ -0,0 +1,47 @@
+package com.enonic.lib.guillotine;
+
+import java.util.function.Supplier;
+
+import com.enonic.lib.guillotine.macro.HtmlAreaProcessedResult;
+import com.enonic.lib.guillotine.macro.ProcessHtmlParams;
+import com.enonic.lib.guillotine.macro.ProcessHtmlService;
+import com.enonic.lib.guillotine.mapper.HtmlAreaResultMapper;
+import com.enonic.xp.portal.PortalRequest;
+import com.enonic.xp.script.ScriptValue;
+import com.enonic.xp.script.bean.BeanContext;
+import com.enonic.xp.script.bean.ScriptBean;
+
+public class ProcessHtmlHandler
+    implements ScriptBean
+{
+    private Supplier<ProcessHtmlService> processHtmlServiceSupplier;
+
+    private PortalRequest request;
+
+    @Override
+    public void initialize( final BeanContext context )
+    {
+        this.request = context.getBinding( PortalRequest.class ).get();
+        this.processHtmlServiceSupplier = context.getService( ProcessHtmlService.class );
```

Comment:
OSGi service inside lib is a bad practice.
Any app with this lib will register a bogus service.

Reply:
done